tck/src/main/java/org/eclipse/sparkplug/tck/test/TCKTest.java

```diff
@@ -43,20 +43,26 @@
  */
 public abstract class TCKTest {
 
-    protected final @NotNull Map<String, String> testResults = new TreeMap<>();
+	private static final @NotNull Logger logger = LoggerFactory.getLogger("Sparkplug");
+	protected final @NotNull Map<String, String> testResults = new TreeMap<>();
 
-    private static final @NotNull Logger logger = LoggerFactory.getLogger("Sparkplug");
+	public Utils.Profile getProfile() {
+		return profile;
+	}
 
-    public void onMqttConnectionStart(ConnectionStartInput connectionStartInput) {
+	public void setProfile(Utils.Profile profile) {
+		this.profile = profile;
+	}
 
-    }
+	protected @NotNull Utils.Profile profile;
 
-    public void onAuthenticationSuccessful(AuthenticationSuccessfulInput authenticationSuccessfulInput) {
+	public void onMqttConnectionStart(ConnectionStartInput connectionStartInput) {
+	}
 
-    }
+	public void onAuthenticationSuccessful(AuthenticationSuccessfulInput authenticationSuccessfulInput) {
```

Comment:
I still need the monitor to be called for the these and other events when the profile is not broker.

And connect, disconnect, subscribe and publish should not call the Monitor if the current profile is broker, I think,

Reply:
added the condition for the callbacks